libs/language-server/src/lib/lsp/jayvee-scope-computation.ts

```diff
@@ -0,0 +1,52 @@
+// SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+
+import {
+  type AstNode,
+  type AstNodeDescription,
+  DefaultScopeComputation,
+  type LangiumCoreServices,
+  type LangiumDocument,
+} from 'langium';
+
+import {
+  isBuiltinConstrainttypeDefinition,
+  isConstraintDefinition,
+  isIotypeDefinition,
+  isReferenceableBlockTypeDefinition,
+  isTransformDefinition,
+  isValuetypeDefinition,
+} from '../ast';
+
+export class JayveeScopeComputation extends DefaultScopeComputation {
+  constructor(services: LangiumCoreServices) {
+    super(services);
+  }
+
+  protected override exportNode(
+    node: AstNode,
+    exports: AstNodeDescription[],
+    document: LangiumDocument,
+  ): void {
+    // export the exportable top-level elements
+    if (!this.isExportable(node)) {
+      return;
+    }
+
+    super.exportNode(node, exports, document);
+  }
+
+  isExportable(node: AstNode) {
+    // pipelines are not exported
```

Comment:
If the logic is pipelines are not exported, why not disallow list them instead of allowlisting everything else? Would be better for future additions to the language.

Reply:
I agree, but then comments (hidden nodes) are exported as well.
Then we'd have to find out what to not export first I guess.
The more I think about it, the more reasonable it seems. Will give it a try!

Reply:
I gave it some more thought and came to the conclusion that we should leave it as it is.

Your suggestion makes total sense, but it will be solved by the next update anyway that makes exports explicit (`publish` keyword). So I'd not spend more effort than necessary as it will be reworked anyway.